### DISCUSSION TITLE
Automating OpenModelica Simulations using OpenModelica Script

### DISCUSSION THREAD

**Discussion by lenri001** [OpenModelica/OpenModelica#12308] [Q&A]
I am trying to automate my openModelica simulations. My three goals are as follows:
1.     Run a for loop to run the different simulations
2.     Change the different parameters needed for each simulation with each parameter is described an array
3.     Save the output of the siumlation as a csv file, but with only a few parameters selected (Filesize would be too large if all parameters were selected)

This is the current OMS I have:
`
loadFile("scenario_1_fists_paper_microgrid_no_bess_no_solar_l2_l3_no_ev_charging.mo"); getErrorString();
parameter Real BESSCapacity[3] = {1620000000, 3240000000, 16200000000}; getErrorString();
parameter String BESSFunction[3] = {"demandResponse", "demandResponse", "peakShaving"}; getErrorString();
equation
  for i in {1, 2, 3} loop
    \\ Do something for parameter sweep
    simulate(conf_paper_microgrid, fileNamePrefix = "run_1", outputFormat = "csv"); getErrorString();
  end for

`
I am not sure if for loops in openmodelica script are the same as in openmodelica.  I am new to openmodelica scripting, so apologies if some of these questions are trivial. Thank you for the help. I posted this question on stackoverflow as well https://stackoverflow.com/questions/78376146/automating-openmodelica-simulations-using-openmodelica-script.

**Comment by casella** [comment]
@albertoleva may help you with that

**Comment by lucas-mior** [comment]
openmodelica scripting is very difficult to use. I suggest you to wrap your simulation parameters in a more powerful scripting language and only call the bare minimum of oms code.

**Reply by casella** [reply]
A popular option is to use Python, accessing OMC via [OMPython](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/ompython.html#ompython-openmodelica-python-interface)
